code/game/objects/effects/anomalies.dm

```diff
@@ -482,12 +488,15 @@
 		else
 			O.ex_act(EXPLODE_HEAVY)
 
-/obj/effect/anomaly/bhole/proc/grav(r, ex_act_force, pull_chance, turf_removal_chance)
-	for(var/t = -r, t < r, t++)
-		affect_coord(x + t, y - r, ex_act_force, pull_chance, turf_removal_chance)
-		affect_coord(x - t, y + r, ex_act_force, pull_chance, turf_removal_chance)
-		affect_coord(x + r, y + t, ex_act_force, pull_chance, turf_removal_chance)
-		affect_coord(x - r, y - t, ex_act_force, pull_chance, turf_removal_chance)
+/obj/effect/anomaly/bhole/proc/grav(radius = 0, ex_act_force, pull_chance, turf_removal_chance)
+	if(!radius || prob(25)) // Base 25% chance of not damaging any floors or pulling mobs
+		return
+
+	for(var/t in -radius to (radius - 1))
```

Comment:
Could you maybe make this pull turfs in the radius by range() or block instead of using coords?

Reply:
I've tried, but it's interestingly done.
It pulls turfs in an `X` shape, and that's not really more performant to do with `block()` or `range()` magic

Reply:
That's fair! I'm not all that worried about performance because coord math is just fine in that regard.